### PR TITLE
feat(donor-wall): improve donor wall performance and fix existing issues #3312

### LINKED ISSUE #3312: feat(donor-wall): improve donor wall performance and fix existing issues
### 1. Improve query performance

Each time the grid is queried I see an excessive number of queries.

### 2. Comments are not saving or displaying properly

When you add a comment to the donation it does not display in the grid or in the admin.

### 3. The comment metabox on the donation details needs to allow the comment to be edited. So it should display the comment in a text area regardless if there is a donation or not.

Make the donation comment editable by making this field a text area. Therefore admins can edit or go back and add/edit comments for donations.

### 4. Support `comment_length` - The number of words to display for donation comments before a "Read more" field displays.

This is the number of words a donation comment will display be truncated before being cut off with a "Read more" text.

### 5. Support "Read more" text opening the entire comment in a modal window.

When someone clicks on the "Read more" text of a truncated column a give modal window should open with the entire comment for them to read.

### 6. Support customizing the "Read more" text via the `readmore_text` attribute

If an admin want to customize the "Read more" text for truncated comments they can do that by passing any string to this new attribute.

### 7. Support the `only_comments` bool attribute - this will only display donors with comments

If an admin only wants to display donors with comments then this will allow filtering the donor that do not have comments.

### 8. Support `donation_amount` and `donation_count` options for the `orderby` attribute

Admins need to be able to adjust the order of the donors by the total _count_ of their donations or the total _amount_. 

**Count** = 12 donations, 10 donations, 5, donations, 1 donations 

**Amount** = $1,000 $500 $25

Note: if the _counts_ the same (1, 1, 1,) then the order of those will be by latest donation by date _first_.

### 9. AJAXify the pagination so it doesn't have to have a page reload

Nearly all donor walls on sites like gofundme use AJAX rather than page reloads for pagination. This should apply to ours as well.

## ToDo
- [x] Point 1 (removed duplicate quesy issue)
- [x] Point 2
- [x] Point 3
- [x] Point 4
- [x] Point 5
- [x] Point 6
- [x] Point 7
- [x] Point 8
- [x] Point 9
- [x] Test

### PR DESCRIPTION
**Pull request by ravinderk**
This pr will resolve #3312

### CODE REVIEW

**Comment by ravinderk** [comment]
Slack discussion with @DevinWalker :

1. Sure if it’s more performing
2. Sure

File: includes/admin/upgrades/upgrade-functions.php [hunk 1]
```diff
@@ -372,6 +372,15 @@ function give_show_upgrade_notices( $give_updates ) {
 			'depend'   => array( 'v20_move_metadata_into_new_table' ),
 		)
 	);
+	
+	// v2.1.5 Add additional capability to the give_manager role.
```
Comment (mehul0810) [inline includes/admin/upgrades/upgrade-functions.php]: @ravinderk I think this update should be related to `v2.2` and not `v2.1.5` as I can see the PR to be assigned to be merged in release `2.2`. Please correct me, if it is intentional.

Comment (ravinderk) [reply includes/admin/upgrades/upgrade-functions.php]: @mehul0810 this code is related to `2.1.5` which appearing here because of merge conflict.

**Comment by raftaar1191** [comment]
## Finding
1. Support the only_comments bool attribute - this will only display donors with comments
-> Not working


2. Support donation_amount and donation_count options for the orderby attribute
-> Not tested 

3. AJAXify the pagination so it doesn't have to have a page reload
-> Not tested 


4. Can see same option in a drop down after running `npm run dev`
image

**Comment by raftaar1191** [comment]
@ravinderk 

When new donor try to make a new donation and also create an account the donation comment is not getting added 


image


Video Link: https://screencast-o-matic.com/watch/cF1I6fF1h3

**Comment by ravinderk** [comment]
@raftaar1191 Above issue has been resolved. You can review it.

**Comment by ravinderk** [comment]
@raftaar1191 please review again with latest changes as we discussed over call

**Pull request merged**